File: 프로그래머스_고득점kit/Hash/위장.py
# [[의상이름, 의상종류]] 주어지면
# 조합의 수(최소 1 의상 선택, 같은 종류의 옷 중에서 1/0개씩)
# (2+1)(1+1) - 1 : 종류별 1 or nothing 곱한 다음 - nothing
def solution(clothes:list):
    answer = 1
    dic = dict()
    for _, kind in clothes:
        if kind in dic:
            dic[kind] += 1
        else:
            dic[kind] = 2
            
    for el in dic:
        answer *= dic[el]
    
    return answer-1

solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]])

File: 프로그래머스_고득점kit/Hash/위장.java
// 종류당 1/0개 입을 수 있고, 전체 중 하나는 입어야 함
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();
        for (String[] clo : clothes) {
            map.put(clo[1], map.getOrDefault(clo[1], 1)+1);
        }

        for (Integer v : map.values()){
            answer *= v;
        }
        return answer-1;
    }
}